projects/scripts/adi_board.tcl

```diff
@@ -335,6 +335,199 @@ proc ad_xcvrpll {m_src m_dst} {
+  ad_connect  $name/rate_gen/pulse_period  GND
+  ad_connect  $name/rate_gen/load_config  GND
+  ad_connect  $name/active  $name/rate_gen/rstn
+
+  # add filter instances for n channels
+  for {set i 0} {$i < $n_chan} {incr i} {
+    ad_ip_instance fir_compiler $name/${filter_name}_${i} [ list \
+      Decimation_Rate 1 \
+      Filter_Type "Interpolation" \
+      Interpolation_Rate $filter_rate \
+      Number_Paths $parallel_paths \
+      Clock_Frequency $core_clk_mhz \
+      Sample_Frequency $sampl_freq_mhz \
+      CoefficientSource COE_File \
+      Coefficient_File $coe_file \
+      Coefficient_Fractional_Bits 0 \
+      Data_Fractional_Bits 15 \
+      Coefficient_Sets 1 \
+      Coefficient_Sign Signed \
+      Coefficient_Structure Inferred \
+      Coefficient_Width 16 \
+      ColumnConfig 5 \
+      Filter_Architecture Systolic_Multiply_Accumulate \
+      Number_Channels 1 \
+      Output_Rounding_Mode Symmetric_Rounding_to_Zero \
+      Output_Width 16 \
+      Quantization Integer_Coefficients \
+      RateSpecification Frequency_Specification \
+      Zero_Pack_Factor 1
+    ]
+
+    ad_connect $name/aclk $name/${filter_name}_${i}/aclk
+
+    create_bd_pin -dir I $name/dac_valid_$i
+    create_bd_pin -dir I $name/dac_enable_$i
+    create_bd_pin -dir O $name/valid_out_$i
+    create_bd_pin -dir O $name/enable_out_$i
+    create_bd_pin -dir I -from [expr 16*$parallel_paths-1] -to 0 $name/data_in_$i
+    create_bd_pin -dir O -from [expr 16*$parallel_paths-1] -to 0 $name/data_out_$i
+
+    ad_ip_instance util_vector_logic $name/logic_and_$i [list \
+      C_SIZE 1]
+
+    create_bd_cell -type module -reference ad_bus_mux $name/out_mux_$i
+    set_property -dict [list \
+      CONFIG.CH_DW [expr 16 * $parallel_paths]] [get_bd_cells $name/out_mux_$i]
+
+    ad_connect  $name/rate_gen/pulse  $name/logic_and_$i/Op1
+    ad_connect  $name/dac_valid_$i  $name/logic_and_$i/Op2
+    ad_connect  $name/logic_and_$i/Res  $name/${filter_name}_${i}/s_axis_data_tvalid
+    ad_connect  $name/${filter_name}_${i}/s_axis_data_tdata  $name/data_in_$i
+
+    ad_connect  $name/rate_gen/pulse  $name/out_mux_${i}/valid_in_1
+    ad_connect  $name/dac_enable_$i  $name/out_mux_${i}/enable_in_1
+    ad_connect  $name/${filter_name}_${i}/m_axis_data_tdata  $name/out_mux_${i}/data_in_1
+
+    ad_connect  $name/dac_valid_$i  $name/out_mux_${i}/valid_in_0
+    ad_connect  $name/dac_enable_$i  $name/out_mux_${i}/enable_in_0
+    ad_connect  $name/data_in_$i  $name/out_mux_${i}/data_in_0
+
+    ad_connect  $name/out_mux_${i}/valid_out  $name/valid_out_$i
+    ad_connect  $name/out_mux_${i}/enable_out  $name/enable_out_$i
+    ad_connect  $name/out_mux_${i}/data_out  $name/data_out_$i
+
+    ad_connect  $name/active  $name/out_mux_${i}/select_path
+  }
+}
+
+###################################################################################################
+###################################################################################################
+
```

Comment:
Can we create a single process, like add_filter? And you just define by its parameters if it will be a DDC or DUC.

Reply:
There are some differences between the two, I prefer to keep them separate for now.

Reply:
OK, no problem.